src/HealthChecks.Nats/NatsHealthCheck.cs

```diff
@@ -0,0 +1,100 @@
+﻿namespace HealthChecks.Nats
+{
+    using Microsoft.Extensions.Diagnostics.HealthChecks;
+    using System.Text;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using NATS.Client;
+    using System.Collections.Generic;
+    using System;
+
+    /// <summary>
+    /// Health check for Nats Server
+    /// </summary>
+    public class NatsHealthCheck : IHealthCheck
+    {
+        /// <summary>
+        ///  Provides factory methods to create connections to NATS Servers.
+        /// </summary>
+        private readonly ConnectionFactory _connectionFactory = new ConnectionFactory();
+        
+        private readonly NatsOptions _options;
+
+        /// <summary>
+        /// An NATS.Client.IConnection object connected to the NATS server.
+        /// </summary>
+        private IConnection _connection = null;
+
+        public NatsHealthCheck(NatsOptions natsOptions) => _options = natsOptions;
+
+        public Task<HealthCheckResult> CheckHealthAsync(
+            HealthCheckContext context,
+            CancellationToken cancellationToken = default)
+        {
+            try
+            {
+                _connection = CreateConnection(_options);
+            }
+            catch (Exception ex)
+            {
+                var unhealthy = HealthCheckResult.Unhealthy(ex.Message, ex, new Dictionary<string, object> { ["type"] = ex.GetType().Name });
+                return Task.FromResult(unhealthy);
+            }
+
+            if (_connection is null) return Task.FromResult(HealthCheckResult.Unhealthy("null connection"));
```

Comment:
Worth to move this line inside `try` after  ` _connection = CreateConnection(_options);` and throw if null.

Reply:
Or even inside `CreateConnection`.

Reply:
Commit: b1e6ad14cadedf1eaa0e0d21722551d351167da6 [b1e6ad1]
Parents: c839f8fa6a
PR 891 disposable, editor config